hack/ci-e2e-kind.sh

```diff
@@ -20,7 +27,7 @@ fi
 
 source "${REPO_ROOT}"/gardener/hack/ci-common.sh
 
-clamp_mss_to_pmtu
```

Comment:
Is it no longer needed?

Reply:
This was actually a leftover as I had removed the corresponding function but noticed it still laying around until now. 
Initially, I had went through the https://github.com/kubernetes/test-infra/issues/23741 and my understanding was that the issue is fixed and the work around is no longer necessary. 

This [comment](https://github.com/kubernetes/test-infra/issues/23741#issuecomment-960211068) from the original issue makes me a bit hesitant to also remove the workaround in our other repos, though.

> Workaround has been disabled but is available to re-enable via config change if needed